### PR TITLE
fix: ILIKE operator not supported in sqlite

### PR DESCRIPTION
**Pull request by zakariaelas**
### What
- We have a bunch of queries in our codebase that use the ILIKE operator, which is not supported by sqlite. This PR introduces a fix (#374) for those queries (except one)
### Why
- Make the on-boarding more polished by preventing such errors
### How
- Used the `ILike()` typeorm function which handles drivers that don't support the operator 
### Limitations
- Currently, the `list` method of `productService` will still throw an error. The reason is that typeorm's `where()` method which allows to specify the `ILike()` function [does not work](https://github.com/typeorm/typeorm/pull/7805) for `@OneToMany` relationships. As we search on the product variants' sku and title , which is essentially a `@OneToMany` relationship, we can't use `ILike()` . Given that sqlite will not be used in prod, and that we don't want to clutter the service layer with noise, the PR simply ignores this corner case.

### CODE REVIEW

File: packages/medusa/src/services/product-variant.js [hunk 1]
```diff
@@ -558,14 +558,11 @@ class ProductVariantService extends BaseService {
...
-            qb.where([
-              { sku: Raw(a => `${a} ILIKE :q`, { q: `%${q}%` }) },
-              { title: Raw(a => `${a} ILIKE :q`, { q: `%${q}%` }) },
-            ]).orWhere(`product.title ILIKE :q`, { q: `%${q}%` })
-          })
-        )
+        qb.where(where).andWhere([
+          { sku: ILike(`%${q}%`) },
+          { title: ILike(`%${q}%`) },
+          { product: ILike(`%${q}%`) },
```
Comment (olivermrbl) [inline packages/medusa/src/services/product-variant.js:564]: Should this not be `{ product: { title: ILike(`%${q}%`) } }`? Just looking at the other ones above :)

Comment (olivermrbl) [reply packages/medusa/src/services/product-variant.js:564]: And then delete the inner join?

Comment (zakariaelas) [reply packages/medusa/src/services/product-variant.js:564]: Nice catch! Sorry, will fix now!

Comment (olivermrbl) [reply packages/medusa/src/services/product-variant.js]: We should consider adding a couple of integration tests to ensure it functions :)

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**